pkg/csi/cinder/nodeserver.go

```diff
@@ -526,18 +527,31 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
 	}
 	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume path not provided")
+	}
 
-	args := []string{"-o", "source", "--first-only", "--noheadings", "--target", volumePath}
-	output, err := ns.Mount.Mounter().Exec.Command("findmnt", args...).CombinedOutput()
+	_, err := ns.Cloud.GetVolume(volumeID)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not determine device path: %v", err)
-
+		if cpoerrors.IsNotFound(err) {
+			return nil, status.Error(codes.NotFound, fmt.Sprintf("Volume with ID %s not found", volumeID))
+		}
+		return nil, status.Error(codes.Internal, fmt.Sprintf("NodeExpandVolume failed with error %v", err))
 	}
-	devicePath := strings.TrimSpace(string(output))
 
-	if devicePath == "" {
-		return nil, status.Error(codes.Internal, "Unable to find Device path for volume")
+	output, err := ns.Mount.GetMountRefs(volumePath)
```

Comment:
Is the o/p of the command same as `findmnt` which is used before? Does this func solve any particular issue?

Reply:
@jichenjc 2 things actually , which I need to reconfirm before further review
1. sanity test runs across fake cloudprovider , doesnt do actual calls to openstack. So, I would like to know if the expand functionality is tested again after this change (as CI doesnt cover the functionality.)
2. Sanity test passed after this change because of the line it is added here https://github.com/kubernetes/cloud-provider-openstack/pull/1496/files#diff-7ee33cf89d8ea95985dc055c632a7c1a8f9570a6b9bbfe3b1171e98a2dc55d49R71 , which doesnt do actual operation , its fake mounter , So I want to know if the result returned by the function is same as what we expect.

Reply:
ok, #2 is something I didn't know ,thanks for let me know this 
I will dig further to know what's there, appreciate the info